### PR TITLE
Feature/headless

### PR DESCRIPTION
**Pull request by heritechie**
Add headless module using puppeteer on lib folder, for page that need to rendered to scrap

### CODE REVIEW

File: lib/headless.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  const page = await browser.newPage();
+  console.log(`Open ${url}`)
+  await page.goto(url);
+
+  let saveLink = await page.evaluate(() => document.querySelector('.download-link a').href)
+  
+  await page.goto(saveLink);
+
+  return new Promise( (resolve) => {
+    setTimeout(async () => {
```
Comment (adekiruz) [inline lib/headless.js:18]: I think this part still can be improved to not using timeout

File: index.js [hunk 2]
```diff
@@ -3,16 +3,16 @@ const figlet = require("figlet");
...
 process.stdout.write("\033c");
-// (async () => {
-//   const link = await getDownloadLink(
-//     "https://www.samehadaku.tv/bem-episode-3/"
-//   );
-//   console.log(link);
-// })();
-
-// print figlet text
+(async () => {
```
Comment (adekiruz) [inline index.js]: I think this part is no longer needed

File: index.js [hunk 3]
```diff
@@ -3,16 +3,16 @@ const figlet = require("figlet");
...
 // clean the console
 process.stdout.write("\033c");
-// (async () => {
-//   const link = await getDownloadLink(
-//     "https://www.samehadaku.tv/bem-episode-3/"
-//   );
-//   console.log(link);
-// })();
-
-// print figlet text
```
Comment (adekiruz) [inline index.js]: this comment for describe code please dont remove

**Review by adekiruz — COMMENTED** [review]
Its great!

File: lib/headless.js (hunk 1 above)
Comment (heritechie) [reply lib/headless.js:18]: on progress ..
thanks for review

**Pull request merged**